rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+
+This is not the path we are going to pursue. It feels important to say that up front.
+
+But before we go deeper into where we are headed, it's worth looking at the implied dichotomy between "decentralization" and product features. That this dichotomy is almost reflexively referred to likely results from failing to appropriately frame "decentralization" in terms of product features. So let's start by framing the product features we want to unlock through decentralization.
+
+## The Network We Want
+
+### No Data Loss
+
+1. Users should be able to specify how many unique copies of their data they want backed up, and the W3S network should handle insuring that many copies remain backed up as long as the user pay to store in W3S.
+
+### Perpetuity
+
+1. Users should be able to store hot data in W3S that will be available for  retrieval until they stop paying for it. They should not have to worry about any single business entity holding their data disappearing. If it does, the data should go somewhere else without their needing to intervene.
+
+### Sensible defaults with customization
+
+1. Users should have the option to store with simply "the W3S network" rather than a specific business entity or Filecoin SP. This is a hard lesson learned from Filecoin that forcing users to choose from a marketplace where information on available options is hard to come by is a terrible user experience.
+
+2. Users should still have the option to choose a provider they like, or to move between providers.
+
+### Trust
+
+In general, users should be able to "trust" W3S to handle their data well and meet service level agreements. That trust should be derived by giving users the ability to verify any actions taken on the W3S network. 
+
+1. Users should be able to gather a detailed history of what has happened with their data when they put it on the W3S Network.
+
+2. A user paying for storage should be able to verify that data is being stored.
+
+3. A user paying for storage should be able to verify it can be retrieved quickly. 
+
+4. A user paying for multiple backups should be able to verify each backup is still functioning.
+
+5. If verification demonstrates that a particular provider is failing to provide a service that the user has paid for, the network should detect the failures and move their data automatically. (or alternatively, the user should have the ability to create a dispute and move to a different provider while getting money back)
+
+### Distribution of resources
+
+6. Users should not have to worry about large service providers establishing monopolies that reduce user choice or drive up costs. To insure this, W3S should emphasize data portabiltity, fairness in data distribution, and low barrier to entry for new providers.
+
+## But who wants this?
+
+A common refrain in response to some of the positive properties that "decentralization" might enable is "none of our existing users asked for this". 
```

Comment:
Speaking for myself, the primary thing that I don't think anyone is asking for is "I want the bits of my upload to be stored in hot storage hosted by random participants in the w3s network." 

I'm _very_ sympathetic to the idea that people don't always want to think about where their bits actually live, so I do think we should provide a "default" storage location and I don't necessarily think the healthiest way to build that is to have us (eg, w3s-maintained S3 or equivalent) be the one and only default, BUT I _do_ think that things get a lot lot lot more complicated and risky when we try to rely on an incentivized permissionless network for this, imho in a way that is out of proportion to the amount of apparent market demand for this specific property.

Data portability, verifiability of actions taken on the network, sensible defaults with the ability to customize, storage perpetuity - all of these things seem (to me) clearly valuable and desired by our customers and the market at large, but I don't think any of them require a part of our system that can distribute the actual bits to a random group of w3up operators. The bits that might utilize this - a desire to avoid becoming a monopoly (defacto or otherwise), resiliance in the face of a catastrophic failure of the w3s organization - do seem valuable to me, but I'm not convinced that "a permissionless network of hot storage nodes as the default destination for the bits" is the simplest or most elegant solution to either of these. Maybe I'm wrong - maybe this is the only way to achieve these goals! - but I would definitely like to spend more time kicking this around before we go down this road, because it feels like a very very risky path to take.

Reply:
eventually, I see a heterogenous network that employs a mix of permissionless nodes as well as more trusted gaurantors of service quality. At minimum, if a user gives us data, we shouldn't hand it out randomly -- we should hand it to those who have the best reputation.

I've noticed there's a certain assumption here of "we'll build it like Saturn" which is why I wanted to specifically address that below. I absolutely agree it's silly to scale to a huge permissionless network until there is demand. I call out the lack of economic connection between supply and demand to be perhaps Saturn's biggest mistake. At the same time, I don't personally think an entirely federated system is the end state. Does it look like Saturn? Probably not. But I do want us to get to a world where anyway can be part of it, and at least take a shot at offering a better service for a lower price, whether they are large or small.

Reply:
what I'd like to see is:

1) we set up this network and serve as the default point of entry for people
2) we continually move to make it possible for people to decouple themselves from w3s-the-legal-entity and make this a core value proposition of our service - we will constantly need to balance this against the needs and desires of stakeholders in w3s-the-legal-entity so we'll need to be thoughtful about how we maintain this line
3) in the long run steady state, w3s-the-legal-entity strives to continually improve the "default experience" for engaging with w3s-the-network while simultaneously encouraging an ecosystem of businesses to grow around w3s-the-network - some of these businesses may be addressing market segments that we choose not to engage in, while others may directly compete with us - this is a feature and something we will ideally make a core value proposition of w3s-the-network

Reply:
re: 3

Yes, this how I think of it.

For me the end state of the w3s entity is some kind of foundation where a small part of protocol revenue pays a core team to keep developing the network.

The nice thing is once the token is liquid, if we end up with a token and a token raise, the VCs have already gotten their exit. This to me is the moment you can transition to a model that more closely resembles https://github.com/web3-storage/RFC.staging/pull/1 . This is all of course fantasy, based on everything going exactly how it's supposed to, not getting caught up in some kind of crypto bubble, but that actually is how I would like it to go.